internal/xdscache/v3/endpointslicetranslator.go

```diff
@@ -61,7 +61,7 @@ func (c *EndpointSliceCache) RecalculateEndpoints(port, healthPort v1.ServicePor
 					continue
 				}
 
-				if p.Protocol == nil || (healthPort.Protocol != *p.Protocol || port.Protocol != *p.Protocol) && *p.Protocol != v1.ProtocolTCP {
+				if p.Protocol == nil || healthPort.Protocol != *p.Protocol || port.Protocol != *p.Protocol || *p.Protocol != v1.ProtocolTCP {
```

Comment:
I'd like to see this split up into a couple different if statements, perhaps one checking for nil, one checking for non-TCP, and one checking for non-matching protocols on the port or healthPort. Same for endpointstranslator. Thanks!

Reply:
hello @skriss 

Do you mean this way ?
```Go
	if p.Protocol == nil {
		if *p.Protocol != v1.ProtocolTCP {
			if healthPort.Protocol != *p.Protocol || port.Protocol != *p.Protocol {

			}

		}

	}

```

This way chage the "||" logic

Reply:
I was thinking like:

```go
if p.Protocol == nil {
        continue
}

if *p.Protocol != v1.ProtocolTCP {
        continue
}

// similarly, check if *p.Protocol matches port or healthPort protocol
```

Reply:
Is there some differences between one statement with "||" and splitting up into a couple different if statements ? IMO,it's the same

Reply:
IMO it improves readability to split up and have several smaller bits of early-exit logic. Of course, logically they are the same, but the issue was about readability.